src/utils/daos_hdlr.c

```diff
@@ -2948,6 +2963,456 @@ fs_copy_hdlr(struct cmd_args_s *ap)
+	d_sg_list_t	sgl;
+	d_iov_t		iov;
+	int		rc;
+
+	/* set sgl values */
+	sgl.sg_nr     = 1;
+	sgl.sg_nr_out = 0;
+	sgl.sg_iovs   = &iov;
+	d_iov_set(&iov, buf, buf_len);
+
+        rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, dkey, 1, iod, &sgl,
+			    NULL, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "failed to fetch source value: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+        rc = daos_obj_update(*dst_oh, DAOS_TX_NONE, 0, dkey, 1, iod,
+			     &sgl, NULL);
+	if (rc != 0) {
+		fprintf(stderr, "failed to update destination value: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+out:
+	return rc;
+}
+
+static int
+cont_copy_recx_array(daos_key_t *dkey,
+		     daos_key_t *akey,
+		     daos_handle_t *src_oh,
+		     daos_handle_t *dst_oh,
+		     daos_iod_t *iod)
+{
+	int			rc = 0;
+	int			i = 0;
+	uint64_t		buf_len = 0;
+	uint32_t		number = 5;
+	daos_anchor_t		recx_anchor = {0}; 
+	d_sg_list_t		sgl;
+	d_iov_t			iov;
+	daos_epoch_range_t	eprs[5];
+	daos_recx_t		recxs[5];
+	daos_size_t		size;
+
+	while (!daos_anchor_is_eof(&recx_anchor)) {
+		/* list all recx for this dkey/akey */
+		rc = daos_obj_list_recx(*src_oh, DAOS_TX_NONE, dkey,
+			akey, &size, &number, recxs, eprs, &recx_anchor,
+			true, NULL);
+		if (rc != 0) {
+			fprintf(stderr, "failed to list recx: %d\n", rc);
+			D_GOTO(out, rc);
+		}
+
+		/* if no recx is returned for this dkey/akey move on */
+		if (number == 0) 
+			continue;
+		for (i = 0; i < number; i++) {
+			buf_len = recxs[i].rx_nr;
+		        char        buf[buf_len];
+
+			/* set iod values */
+			(*iod).iod_type  = DAOS_IOD_ARRAY;
+			(*iod).iod_size  = 1;
+			(*iod).iod_nr    = 1;
+			(*iod).iod_recxs = &recxs[i];
+
+			/* set sgl values */
+			sgl.sg_nr     = 1;
+			sgl.sg_nr_out = 0;
+			sgl.sg_iovs   = &iov;
+
+			d_iov_set(&iov, buf, buf_len);
+
+			/* fetch recx values from source */
+                        rc = daos_obj_fetch(*src_oh, DAOS_TX_NONE, 0, dkey,
+					    1, iod, &sgl, NULL, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to fetch source "
+					"recx: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+
+			/* update fetched recx values and place in
+			 * destination object
+			 */
+                        rc = daos_obj_update(*dst_oh, DAOS_TX_NONE, 0, dkey,
+					     1, iod, &sgl, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to update destination "
+					"recx: %d\n", rc);
+				D_GOTO(out, rc);
+			}
+		}
+	}
+out:
+	return rc;
+}
+
+static int
+cont_copy_list_akeys(daos_handle_t *src_oh,
+		     daos_handle_t *dst_oh,
+		     daos_key_t diov)
+{
+	int		rc = 0;
+	int		j = 0;
+	int		buf_len = 0;
+	char*		ptr;
+	daos_anchor_t	akey_anchor = {0}; 
+	daos_key_desc_t	akey_kds[ENUM_DESC_NR] = {0};
+	uint32_t	akey_number = ENUM_DESC_NR;
+	char		akey_enum_buf[ENUM_DESC_BUF] = {0};
+	char		akey[ENUM_KEY_BUF] = {0};
+	d_sg_list_t	sgl;
+	d_iov_t		iov;
+	daos_key_t	aiov;
+	daos_iod_t	iod;
+
+	/* loop to enumerate akeys */
+	while (!daos_anchor_is_eof(&akey_anchor)) {
+ 		memset(akey_kds, 0, sizeof(akey_kds));
```

Comment:
(style)  code indent should use tabs where possible
(style)  please, no space before tabs